WorkflowTesting/Sources/RenderTesterResult.swift

```diff
@@ -41,6 +43,22 @@ public struct RenderTesterResult<WorkflowType: Workflow> {
         return self
     }
 
+    /// Exhaustive state testing against the initial state.
+    /// - Parameters:
+    ///   - changes: A function that receives the initial state
+    ///   and is expected to mutate it to match the new state.
+    @discardableResult
+    public func assertState(
+        file: StaticString = #file,
+        line: UInt = #line,
+        changes: (inout WorkflowType.State) throws -> Void
+    ) rethrows -> RenderTesterResult<WorkflowType> where WorkflowType.State: Equatable {
```

Comment:
nit: there is an extension below (~L130) where the utilities for equatable states should probably all go

Reply:
- [ ] move to extension